ext/opentelemetry-ext-prometheus/src/opentelemetry/ext/prometheus/__init__.py

```diff
@@ -0,0 +1,150 @@
+
+    def shutdown(self) -> None:
+        REGISTRY.unregister(self._collector)
+
+
+class CustomCollector:
+    """ CustomCollector represents the Prometheus Collector object
+        https://github.com/prometheus/client_python#custom-collectors
+    """
+
+    def __init__(self, prefix: str = ""):
+        self._prefix = prefix
+        self._metrics_to_export = []
+
+    def add_metrics_data(self, metric_records: Sequence[MetricRecord]):
+        self._metrics_to_export.append(metric_records)
+
+    def collect(self):
+        """Collect fetches the metrics from OpenTelemetry
+        and delivers them as Prometheus Metrics.
+        Collect is invoked every time a prometheus.Gatherer is run
+        for example when the HTTP endpoint is invoked by Prometheus.
+        """
+
+        for metric_batch in list(self._metrics_to_export):
+            for metric_record in metric_batch:
+                prometheus_metric = self._translate_to_prometheus(
+                    metric_record
+                )
+                if prometheus_metric:
+                    yield prometheus_metric
+            self._metrics_to_export.remove(metric_batch)
+
+    def _translate_to_prometheus(self, metric_record: MetricRecord):
+        prometheus_metric = None
+        label_values = metric_record.label_set.labels.values()
+        metric_name = ""
+        if self._prefix != "":
+            metric_name = self._prefix + "_"
+        metric_name += sanitize(metric_record.metric.name)
+
+        if isinstance(metric_record.metric, Counter):
+            prometheus_metric = CounterMetricFamily(
+                name=metric_name,
+                documentation=metric_record.metric.description,
+                labels=metric_record.metric.label_keys,
+            )
+            prometheus_metric.add_metric(
+                labels=label_values, value=metric_record.aggregator.check_point
+            )
+
+        elif isinstance(metric_record.metric, Gauge):
+            prometheus_metric = GaugeMetricFamily(
+                name=metric_name,
+                documentation=metric_record.metric.description,
+            )
+            prometheus_metric.add_metric(
+                labels=label_values, value=metric_record.aggregator.check_point
+            )
+
+        # TODO: Add support for histograms when supported in OT
+        elif isinstance(metric_record.metric, Measure):
+            prometheus_metric = UnknownMetricFamily(
+                name=metric_name,
+                documentation=metric_record.metric.description,
+            )
+            prometheus_metric.add_metric(
+                labels=label_values, value=metric_record.aggregator.check_point
+            )
+
+        else:
+            logger.warning(
+                "Unsupported metric type. %s", type(metric_record.metric)
+            )
+        return prometheus_metric
+
+
+_NON_LETTERS_NOR_DIGITS_RE = re.compile(r"[^\w]", re.UNICODE | re.IGNORECASE)
+
+
+def sanitize(key):
```

Comment:
This is called in only one place, maybe put this code directly there instead of leaving something importable at module level?

Reply:
sanitize need to be applied to label names as well, added that missing logic